From: Borislav Petkov <bp@suse.de>
Date: Sun, 17 Dec 2017 16:37:58 +0100
Subject: xen/x86/CPU: Check speculation control CPUID bit
Patch-mainline: Never, SUSE-Xen specific
References: bsc#1068032

... and enable the corresponding flags.

Carved out from a patch by Tim Chen <tim.c.chen@linux.intel.com> and
improved.

After microcode reload, we need to check CPUID directly as we don't
update the X86_FEATURE flags after a reload.

While at it, remove the __cpuinitdata annotation of cpu_caps_set which
is just silly and shuts up an annoying warning. See:

  148f9bb87745 ("x86: delete __cpuinit usage from all x86 files")  

Signed-off-by: Borislav Petkov <bp@suse.de>
Automatically created from "patches.suse/20-x86-cpu-check-speculation-control-cpuid-bit.patch" by xen-port-patches.py

--- a/arch/x86/kernel/cpu/common-xen.c
+++ b/arch/x86/kernel/cpu/common-xen.c
@@ -419,8 +419,8 @@ static const char *__cpuinit table_looku
 	return NULL;		/* Not found */
 }
 
-__u32 cpu_caps_cleared[NCAPINTS] __cpuinitdata;
-__u32 cpu_caps_set[NCAPINTS] __cpuinitdata;
+__u32 cpu_caps_cleared[NCAPINTS];
+__u32 cpu_caps_set[NCAPINTS];
 
 void load_percpu_segment(int cpu)
 {
